deepchem/rl/a3c.py

```diff
@@ -21,12 +21,13 @@ def __init__(self, value_weight, entropy_weight, **kwargs):
     self.entropy_weight = entropy_weight
 
   def create_tensor(self, **kwargs):
-    reward, action, prob, value = [layer.out_tensor for layer in self.in_layers]
-    log_prob = tf.log(prob)
+    reward, action, prob, value, advantages = [layer.out_tensor for layer in self.in_layers]
+    log_prob = tf.log(prob + 0.0001)
     policy_loss = -tf.reduce_sum(
```

Comment:
So isn't this just another value loss?  What is encouraging the network to seed out better rewards/values?  

The way I am reading this loss function right now is it is encouraging certainty -- that is the value loss, policy loss both encourage the value==discounted_rewards.  The entropy also encourages certainty, to not be unsure but to go where you know where you have accurate value function.  

I implemented advantage as seen here
https://medium.com/emergent-future/simple-reinforcement-learning-with-tensorflow-part-8-asynchronous-actor-critic-agents-a3c-c88f72a5e9f2
In order to make the policy loss portion encourage going towards higher rewards.

Still not able to play tic tac toe though.

Am I reading the current implementation of policy loss wrong @peastman ?